### PR TITLE
Add redirect for plans page

### PR DESCRIPTION
**Pull request by RuthShryock**
## Checklist

1. [ ] If you've added code that should be tested, add tests
2. [ ] If you've changed APIs, update (or create!) the documentation
3. [ ] Ensure the tests pass
4. [ ] Make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/master/CONTRIBUTING.md)
5. [ ] Write a title and, if necessary, a description of your work suitable for publishing in our [release notes](https://community.kobotoolbox.org/tag/release-notes)
6. [ ] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes

## Description

If a user, who is not logged in, clicks on a link to the plans page, they will be redirected to the login. Then after logging in, they will be redirected back to the plans page. 

## Notes

Added a [loader](https://reactrouter.com/en/main/route/loader) function to the plans route which calls `authLoader` and redirects the user accordingly based on their authentication status. 

Note: depends on the [react-router version upgrade](https://github.com/kobotoolbox/kpi/pull/4766)

### CODE REVIEW

File: jsapp/js/router/routerUtils.ts
Comment (LMNTL) [inline jsapp/js/router/routerUtils.ts:15]: Only one small issue: when I login after being redirected, instead of sending me to `/#/account/plans`, it sends me back to the login page. It looks like this should be [importing `when` from `'mobx'`](https://mobx.js.org/reactions.html#when) instead of jQuery.

**Review by LMNTL — CHANGES_REQUESTED** [review]
This has my approval once the one issue I found is fixed. Make sure to wait until [kpi#4766](https://github.com/kobotoolbox/kpi/pull/4766) is merged before merging this PR 👍

**Review by LMNTL — APPROVED** [review]

**Comment by tinok** [comment]
If I understand the PR correctly, it allows routing to any page directly (after authentication), not just the plans page? If yes, should the PR name/description be updated to be more generic?

**Comment by RuthShryock** [comment]
@tinok This PR creates a loader function which could be used to redirect any route but for this PR, it is only added the to plans page. So the name should be fine :)

**Pull request merged**
